runtime/bcutil/ROMClassCreationContext.hpp

```diff
@@ -293,7 +290,30 @@ class ROMClassCreationContext
 		}
 	}
 
-	void setIsLambda(bool isLambda) { _isLambda = isLambda; }
+	bool isUnsafeClassShareable() const {
```

Comment:
The name of this function implies it's looking at properties of the class when mostly it's looking at the settings of command line flags.

Maybe rename to `isUnsafeClassSharingEnabled()` and explicitly check the 4 cases mentioned.

ie: `` if (isClassAnon() && vm->anonClassLoader.sharingEnabled) {`` // not the right check but you get the point

Reply:
for anonymous classes, the class loader shared classes enabled flag is set accordingly to ShareAnonymousClasses and this flag is being checked in isROMClassShareable() in the isClassLoaderSharedClassesEnabled() function. So in the function isUnsafeClassSharingEnabled() we don't need to check if ShareAnonymousClasses is enabled when the class is anonymous because if it was disabled, isROMClassShareable() would already return false because of the class loader check. So instead of checking if ShareAnonymousClasses is enabled here, I can add an assertion to assert when it is disabled. Another option could be to remove these lines and enable the class loader flag to be always set:
```
if (J9_ARE_NO_BITS_SET(vm->sharedClassConfig->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_SHAREANONYMOUSCLASSES)) {
	/* if ShareAnonymousClasses is disabled, anonymous classes are not shared so we unset the flag */
	vm->anonClassLoader->flags &= (~J9CLASSLOADER_SHARED_CLASSES_ENABLED);
}
```
and add the check in isUnsafeClassSharingEnabled(). Or I could do both checks but one would be unnecessary in this case.

Reply:
> So instead of checking if ShareAnonymousClasses is enabled here, I can add an assertion to assert when it is disabled.

That's a good approach - it keeps 1 source of truth for this and has this path validate that it's in sync